vnext/Microsoft.ReactNative/IReactDispatcher.h

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+#include "ReactDispatcherHelper.g.h"
+#include <ReactPropertyBag.h>
+#include <dispatchQueue/dispatchQueue.h>
+#include <winrt/Microsoft.ReactNative.h>
+
+namespace winrt::Microsoft::ReactNative {
+
+struct ReactDispatcher : implements<ReactDispatcher, IReactDispatcher> {
+  ReactDispatcher() = default;
+  ReactDispatcher(Mso::DispatchQueue &&queue) noexcept;
+
+  bool HasThreadAccess() noexcept;
+  void Post(ReactDispatcherCallback const &callback) noexcept;
+
+  static Mso::DispatchQueue GetUIDispatchQueue(IReactPropertyBag const &properties) noexcept;
+
+  static IReactDispatcher UIThreadDispatcher() noexcept;
```

Comment:
>UIThreadDispatcher [](start = 26, length = 18)

Should this match api on line 23? Eg: GetUIThreadDispatcher? or the other way around?

Reply:
These are tow different APIs:
- UIThreadDispatcher returns IReactDispatcher associated with the current UI thread.
- GetUIThreadDispatcher is an 'attached property' method that returns UIThreadDispatcher stored in the property bag.

---
In reply to: [423966470](https://github.com/microsoft/react-native-windows/pull/4877#discussion_r423966470) [](ancestors = 423966470)